tests/benchmark.py

```diff
@@ -343,6 +355,94 @@ def generate(
                 break
 
 
+class LlamaCppModelWrapper(ModelWrapper):
+    name: str = "llama_cpp_model_wrapper"  # name of the model wrapper
+    ggml_file_path: str  # path to the ggml model binary path
+    model: Llama  # llama model object
+    tokens: List[llama_token]  # llama tokens
+
+    def __init__(
+        self,
+        tokenizer,
+        max_gen_len,  # n_ctx text context
+        conv_template,
+        ggml_file_path: str,  # path to the ggml model binary path
+        n_parts: int = -1,  # -1 for default
+        n_gpu_layers: int = 0,  # number of layers to store in VRAM
+        seed: int = 1337,  # RNG seed, 0 for random
+        f16_kv: bool = True,  # use fp16 for KV cache
+        logits_all: bool = False,  # the llama_eval() call computes all logits, not just the last one
+        vocab_only: bool = False,  # only load the vocabulary, no weights
+        use_mmap: bool = True,  # use mmap if possible
+        use_mlock: bool = False,  # force system to keep model in RAM
+        embedding: bool = False,  # embedding mode only
+        n_threads: Optional[int] = None,
+        n_batch: int = 512,
+        last_n_tokens_size: int = 64,
+        lora_base: Optional[str] = None,
+        lora_path: Optional[str] = None,
+        verbose: bool = True,
+    ):
+        super().__init__(tokenizer, max_gen_len, conv_template)
+
+        self.name = f"llama_cpp_model_wrapper"
+        self.ggml_file_path = ggml_file_path
+        self.model = Llama(
+            ggml_file_path,
+            n_ctx=max_gen_len,
+            n_parts=n_parts,
+            n_gpu_layers=n_gpu_layers,
+            seed=seed,
+            f16_kv=f16_kv,
+            logits_all=logits_all,
+            vocab_only=vocab_only,
+            use_mmap=use_mmap,
+            use_mlock=use_mlock,
+            embedding=embedding,
+            n_threads=n_threads,
+            n_batch=n_batch,
+            last_n_tokens_size=last_n_tokens_size,
+            lora_base=lora_base,
+            lora_path=lora_path,
+            verbose=verbose,
+        )
+        self.tokens = self.model.tokenize("test".encode("utf-8"))
+        for i in range(2, max_gen_len * 3):
+            # we buffer a token list to avoid the overhead of calling tokenize() for each token
+            # first token is the start token, after that it's all "test" token
+            self.tokens.append(self.tokens[1])
+
+    def sync(self):
+        pass
+
+    def prep_model(self):
```

Comment:
Apology for the lack of the documentation, this `prep_model` is intended to prepare model with the clean kv cache. Can we use this?

Reply:
Yeah sure will move to `prep_model`.